### DISCUSSION TITLE
Does WebSSH support SSH Agent Forwarding? If no, are there any plans to?

### DISCUSSION THREAD

**Discussion by WiggyWare** [isontheline/pro.webssh.net#463] [Q&A] [answered]
Hi, I really love WebSSH. I'm wondering if it currently supports (or will in the future) Agent Forwarding.

Thanks very much in advance, and thanks for a great app!

**Comment by isontheline** [comment, marked as answer]
@WiggyWare Hello Bill👋

Thank you a lot for your support about WebSSH🙏

Unfortunately agent forwarding is not on the roadmap but these are :
* https://github.com/isontheline/pro.webssh.net/issues/266
* https://github.com/isontheline/pro.webssh.net/issues/319
* https://github.com/isontheline/pro.webssh.net/issues/327
* https://github.com/isontheline/pro.webssh.net/issues/327

Agent forwarding is a hard thing to do for me and have some security issues related.

Instead Port Forwarding and "bastion" are a "safer" way to access servers behind a secure SSH host.

What do you think about?

**Reply by x-dragos** [reply]
Hello, i guess your answer is more or less legit for accessing other shells "behind a shell". My use case for agent forwarding is to authenticate with git and github, and i'd rather not put my private key on each server. its kinf of a standard functionality in most other ssh clients (although in most of them it is a "paid" thing... but since i'm guessing most ppl buy WebSSH if they want to use it.. .as i did, they do it because that manage more than 1 server... :) and probably would need something like this.

**Reply by x-dragos** [reply]
have plans changed since 2021 ? :) as i see the ssh_config support doesnt support AgentForward either, there isnt a way to do it for authenticating with git repos aside from coping the private key on the actual server. This would be the last item on the list for me to make this the perfect ssh client for mac. Sad thing is, if this will never be supported, i have to use something else :(

**Reply by isontheline** [reply]
Hi @x-dragos 👋

Unfortunately the Agent Forwarding implementation is not as trivial as I would like.

I will up this thread and issue and make a look at it again.

**Reply by x-dragos** [reply]
Hello @isontheline, 

Thanks for the reply. I never expected it to be an easy thing, nor have i suggested it :) i was just explaoining the main usecase for most of my colleagues and me. I did buy a license for your app, and sadly for me i didnt check this, but i was excited it had everything else i needed. Hopefully this will be implemented at some point, and then i can use it daily and i can uninstall all other ssh managers i am trying out. 

Thanks again for considering this. Hope you decice is worth the effort.

Best regards,
Dragos
